src/transformers/modeling_utils.py

```diff
@@ -505,7 +505,7 @@ def load_state_dict(checkpoint_file: Union[str, os.PathLike]):
         # Check format of the archive
         with safe_open(checkpoint_file, framework="pt") as f:
             metadata = f.metadata()
-        if metadata.get("format") not in ["pt", "tf", "flax"]:
+        if metadata.get("format") not in ["pt", "tf", "flax", "mlx"]:
```

Comment:
This wouldn't be necessary if we replace the `"format"` as discussed in another comment.